File: src/main/java/com/github/bgomar/consolelogger/tools/ConsoleLoggerSettingsConfigurable.java
package com.github.bgomar.consolelogger.tools;

import com.github.bgomar.consolelogger.toolwindow.setup.PropertiesConsoleLoggerToolSetup;
import com.intellij.openapi.options.ConfigurableBase;
import org.jetbrains.annotations.NotNull;

public class ConsoleLoggerSettingsConfigurable extends ConfigurableBase<PropertiesConsoleLoggerToolSetup, ConsoleLoggerSettings> {

  public ConsoleLoggerSettingsConfigurable() {
    super("com.github.bgomar.consolelogger", "ConsoleLogger", "");
  }

  @Override
  protected @NotNull ConsoleLoggerSettings getSettings() {
    return ConsoleLoggerSettings.getInstance();
  }

  @Override
  protected PropertiesConsoleLoggerToolSetup createUi() {
    return new ConsoleLoggerConfigurableUI.ConsoleLoggerConfig(getSettings());
  }
}

File: src/main/kotlin/com/github/bgomar/consolelogger/settings/ConsoleLoggerSettingsConfigurable.kt
package com.github.bgomar.consolelogger.settings

import com.github.bgomar.consolelogger.settings.ConsoleLoggerConfigurableUI
import com.intellij.openapi.options.ConfigurableBase
import com.intellij.openapi.options.ConfigurableUi

class ConsoleLoggerSettingsConfigurable : ConfigurableBase<ConfigurableUi<ConsoleLoggerSettings!>, ConsoleLoggerSettings>("com.github.bgomar.consolelogger", "ConsoleLogger", "") {

  override fun getSettings(): ConsoleLoggerSettings {
    return ConsoleLoggerSettings.instance
  }

  override fun createUi(): ConsoleLoggerConfigurableUI {
    return ConsoleLoggerConfigurableUI(settings)
  }
}